### ISSUE TITLE
PDF Export + Color separation improvement ideas

### ISSUE THREAD

**Issue by Symbian9** [OpenOrienteering/mapper#653]
For now thanks to all its possible export and print with 6 spot color separation option (resulted PDF will have 6 separate page for one per each spot color).
But as for me, its need small improvements:
* Add one more page to the end of PDF with comments and rules how use this document (optional; would be usefull if you give it to publishing company)
* Add printing marks (optional; see how it work in Scribus)
* Color separation for CMYK printing must produce only 4 pages. For now when I try get it result is only 6 spot. How I try do it now:
> Menu -> File -> Export as... -> PDF -> Option -> Color separation + Color mode: Device CMYK (experimental)

Maybe some other future, that I forgot now will add soon.

**Comment by dg0yt** [comment]
Is this request based on experience with / requirements of real print shops?

**Comment by Symbian9** [comment]
@dg0yt, yeah, this is my experience when I try choose one from 10 privat publishing companies.  
1) Also, when map has more than 6 spot colors, then its difficult to remember what is the color order. So, think, its would be good have page inside this pdf where in any time can see what color needed for each page above and what is specific rules for printing.
2) I use OOMapper as simple DTP for city maps and schemes with custom colors, and want print is in CMYK, than need create 4 base spot colors - CYAN, MAGENTA, YELLOW and BLACK. But why not use normally in-build CMYK used for defining colors? Think it must be as when export PDF with  "Color mode: CMYK" then it must produce 4pages based on CMYK column in color window, and use "Color mode: Default" then use values from Spot color column in color window.(because many publishing company have few different prices for printing: print in CMYK,  print in full color or PANTONE)

**Comment by dg0yt** [comment]
I think modern print shops would expect data in proper PDF documents - CMYK as is, and even spot colors by properly marked colors in the document. From my point of view, this is what needs to be implemented next.

The B/W spot color separation export was done as a proof of concept, in connection with the overprinting simulation (which uses the very same principle), and CMYK separations would be a totally different story.

**Comment by Symbian9** [comment]
@dg0yt, I agree that B/W is a trick, but as I sad, this pages currently need one additional page where some info provided about what is difference between all grayscale pages and how print them (in situation when original mapper cant speak with publisher, it will help print all correctly). Imaginate situation, that you forgot what color order on pages was... And now you cant find it because there no any marks on each pages!

Here is example of professional PDF export
* http://sk1project.org/downloads/release_prepress.pdf
* https://facebook.com/308311182521658/photos/a.562085380477569.128234.308311182521658/1163467910339310/

Here is screen of test for approve correct state 
![sK1 2.0 alpha test](https://z-m-scontent-ams3-1.xx.fbcdn.net/hphotos-xfp1/t31.0-8/fr/cp0/e15/q65/12967360_1163467910339310_7497176434405273378_o.png.jpg)
This file created and distributed for testing by developers of sK1 editor for desktop publishing tasks:
* http://github.com/sk1project/sk1-wx

<b>Think, need deeply look inside sK1 sourcecode in section of color management!</b>(maybe some tricks from it will help understand how improve OOMapper)

**Comment by dg0yt** [comment]
The screeshots shows a PDF with one page and 4 plates. This is what I said, not what you requested.

Provide the same preflight check for a orienteering map exported from Mapper in CMYK. Maybe use the "Create Report..." feature.

**Comment by Symbian9** [comment]
I mean, when map will have 9 colors then resulted PDF will include 10 pages=9+1 page with contents of this 9 colors order list and some specific info (maybe comments from map, from "map comment" box).
What a problem export with CMYK value? There on gradients, so think, its must not be hard, right? (I would read soon about colors in PDF).

Also look other fresh examples from sK1project
<quote>While Text On Path is under development we have completed export in PDF (exactly PDF/X-4) with such important features as CMYK colorspace, registration color and SPOT color support (including PANTONE colors), transparency for gradients and images and so on.</quote>
* http://sk1project.org/downloads/pdf_x-4.pdf
* CMYKA - sk1project.org/downloads/pdf_x-4_cmyk.pdf

Details - https://facebook.com/308311182521658/photos/a.562085380477569.128234.308311182521658/1162923570393744/

**Comment by Symbian9** [comment]
@dg0yt, here is some info that I mean about when say "not hard".
* https://forums.adobe.com/thread/997033?tstart=0

Here is some data structure  inside PDF that used for color specific.
So, think, its possible write values from OOMapper using this specification

**Comment by Symbian9** [comment]
@dg0yt, look to this free Python scripts that designed for working with PDF's!
https://github.com/tomcarver/pdf-tools

**Comment by dg0yt** [comment]
Have you ever seen birds? So you know it is possible to fly, and it doesn't look difficult...

Eugeniy, collecting random pieces from the web leads nowhere. I also don't need an explicit mention in every single post. And I start to get tired of discussion your posts rather than Mapper. Even with asking direct questions, it is impossible to know if you want A (CMYK separations on different pages) or B (CMYK plates in a single page).

**Comment by Symbian9** [comment]
Kai, sorry me for links.
So, what I "want in Mapper" is next types of export:
1) Vector graphics - PDF file with CMYK plates in a single page (PDF must be real CMYK color space)
2) Color separation - PDF with few grayscale pages, based on Spot colors names (number of page same as number of spot colors +1 additional page with contents and rules how use it )
3) Raster graphics - export PDF with raster image <b>or</b> raster Image itself, with "Color mode" settings:
* 3.1 - in RGB color mode
* 3.2 - in CMYK color mode

*No need more two export tabs separate "Export -> Image" and "Export -> PDF" submenu's, and only "Export" needed, where option "Raster graphics" is main for that (for export Image way must be "Main menu -> Export -> Raster graphics -> Export... -> File of type: -> PDF or PNG or BMP or TIFF or JPEG")

Think thats all what now Mapper need in this section.
